src/modules/search-result/author.jsx

```diff
@@ -1,28 +1,80 @@
-import React from 'react'
+import React, { useState, useEffect, useRef } from 'react'
 
 import styles from './author.css'
 
+import { Button } from 'elements'
 import { classNames } from 'utils'
 
-const List = ({ children, className, ...htmlProps }) => (
-  <ol className={classNames.use(styles.list).join(className)} {...htmlProps}>
-    {children}
-  </ol>
+const Item = ({ name, className, ...htmlProps }) => (
+  <li
+    itemProp="author"
+    itemScope=""
+    itemType="https://schema.org/Person"
+    className={classNames.use(className)}
+    {...htmlProps}
+  >
+    <a
+      href={`https://core.ac.uk/search?q=author:(${name})`}
+      className={styles.link}
+    >
+      {name.replace(',', ' ')}
+    </a>
+  </li>
 )
 
-const Item = ({ children, href, ...htmlProps }) => {
-  const ElementType = href != null ? 'a' : 'span'
+const List = ({ authors, className, ...htmlProps }) => {
+  const showMoreRef = useRef(null)
+  const [isExpanded, setIsExpanded] = useState(false)
+
+  useEffect(() => {
+    if (isExpanded)
+      showMoreRef.current.parentElement.nextElementSibling.firstChild.focus()
+  }, [isExpanded])
+
+  if (authors.length <= 3) {
+    return (
+      <ol
+        className={classNames.use(styles.list).join(className)}
+        {...htmlProps}
+      >
+        {authors.map((author) => (
+          <Item name={author.name} />
+        ))}
+      </ol>
+    )
+  }
+
   return (
-    <li
-      itemProp="author"
-      itemScope=""
-      itemType="https://schema.org/Person"
-      {...htmlProps}
-    >
-      <ElementType href={href} itemprop="name">
-        {children}
-      </ElementType>
-    </li>
+    <ol className={classNames.use(styles.list).join(className)} {...htmlProps}>
+      <Item aria-setsize={authors.length} name={authors[0].name} />
```

Comment:
You did not use the `aria-posinset`, it's recommended by the spec to combine these 2.

Reply:
Isn't it only required when I want to have a different order of items?